src/issuance_auth.rs

```diff
@@ -0,0 +1,383 @@
+    }
+
+    fn to_bytes(&self) -> Vec<u8> {
+        self.0.to_bytes().to_vec()
+    }
+
+    /// Deserialize the issuance authorization signature from its canonical byte representation.
+    pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        NonZeroScalar::try_from(bytes).ok().map(Self)
+    }
+
+    /// Derives the Orchard-ZSA issuance key for the given seed, coin type, and account.
+    pub fn from_zip32_seed(
+        seed: &[u8],
+        coin_type: u32,
+        account: u32,
+    ) -> Result<Self, zip32::Error> {
+        if account != 0 {
+            return Err(zip32::Error::NonZeroAccount);
+        }
+
+        // Call zip32 logic
+        let path = &[
+            ChildIndex::hardened(ZIP32_PURPOSE_FOR_ISSUANCE),
+            ChildIndex::hardened(coin_type),
+            ChildIndex::hardened(account),
+        ];
+
+        // we are reusing zip32 logic for deriving the key, zip32 should be updated as discussed
+        let &isk_bytes = ExtendedSpendingKey::<zip32::Issuance>::from_path(seed, path)?
+            .sk()
+            .to_bytes();
+
+        Self::from_bytes(&isk_bytes).ok_or(zip32::Error::InvalidSpendingKey)
+    }
+}
+
+impl From<&IssueAuthKey<ZSASchnorr>> for IssueValidatingKey<ZSASchnorr> {
+    fn from(isk: &IssueAuthKey<ZSASchnorr>) -> Self {
+        Self(*schnorr::SigningKey::from(isk.0).verifying_key())
+    }
+}
+
+impl IssueValidatingKey<ZSASchnorr> {
+    pub(crate) fn to_bytes(&self) -> Vec<u8> {
+        self.0.to_bytes().to_vec()
+    }
+
+    fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        VerifyingKey::from_bytes(bytes).ok().map(Self)
+    }
+
+    /// Encodes the issuance validating key into a byte vector, in the manner defined in [ZIP 227][issuancekeycomponents].
+    ///
+    /// [issuancekeycomponents]: https://zips.z.cash/zip-0227#derivation-of-issuance-validating-key
+    pub fn encode(&self) -> Vec<u8> {
+        let ik_bytes = self.to_bytes();
+        let mut encoded = Vec::with_capacity(1 + ik_bytes.len());
+        encoded.push(ZSASchnorr::ALGORITHM_BYTE);
+        encoded.extend(ik_bytes);
+        encoded
+    }
+
+    /// Decodes an issuance validating key from the byte representation defined in [ZIP 227][issuancekeycomponents].
+    ///
+    /// [issuancekeycomponents]: https://zips.z.cash/zip-0227#derivation-of-issuance-validating-key
+    pub fn decode(bytes: &[u8]) -> Result<Self, Error> {
+        if let Some((&algorithm_byte, key_bytes)) = bytes.split_first() {
+            if algorithm_byte == ZSASchnorr::ALGORITHM_BYTE {
+                return Self::from_bytes(key_bytes).ok_or(Error::InvalidIssueValidatingKey);
+            }
+        }
+        Err(Error::InvalidIssueValidatingKey)
+    }
+}
+
+impl IssueAuthSig<ZSASchnorr> {
+    fn to_bytes(&self) -> Vec<u8> {
+        self.0.to_bytes().to_vec()
+    }
+
+    pub(crate) fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        schnorr::Signature::try_from(bytes).ok().map(Self)
+    }
```

Comment:
I think we should remove `to_bytes()` and `from_bytes()` and keep just `encode()` and `decode()` (potentialy `encode_bytes()` and `decode_bytes()`.

I don't want to options to go to bytes. What do you think?
(x3)

Reply:
I agree with you that we will have some confusion if we keep both the to/from_bytes and encode/decode functions.
For IssueAuthSig and IssueValidatingKey, I kept only the encode/decode functions because in the specs, ik and sig are always used with the ALGORITHM_BYTE.
For IssueAuthKey, I kept only the to/from_bytes functions because it is never used with ALGORITHM_BYTE in the spec.

Reply:
We have to update orchard_zsa_issuance_auth_sig.rs test vector file to add ALGORITHM_BYTE 0x00 at the beginning of ik and sig.

Reply:
Updated orchard_zsa_issuance_auth_sig test vectors